streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -92,33 +98,62 @@ public void init(final org.apache.kafka.streams.processor.ProcessorContext conte
         }
 
         @Override
-        public void process(final K key, final V value) {
+        public void process(final Record<KIn, VIn> record) {
             // if the key is null, then ignore the record
-            if (key == null) {
-                LOG.warn(
-                    "Skipping record due to null key. topic=[{}] partition=[{}] offset=[{}]",
-                    context().topic(), context().partition(), context().offset()
-                );
+            if (record.key() == null) {
+                if (context.recordMetadata().isPresent()) {
+                    final RecordMetadata recordMetadata = context.recordMetadata().get();
+                    LOG.warn(
+                        "Skipping record due to null key. "
+                            + "value=[{}] topic=[{}] partition=[{}] offset=[{}]",
```

Comment:
Oh, I'm sorry, but it looks like we need one more revision. Useful as it would be at times, we can't log any data (keys, values, or headers) because it might leak sensitive information into the logs.

Reply:
No worries. This is actually a good findings, as there are other 16 places where we are logging value and keys.  Maybe let's create a issue to handle this in another PR? 

<img width="955" alt="image" src="https://user-images.githubusercontent.com/6180701/126681148-2bd07547-5f56-4cc8-a9f1-88b4826b983e.png">

Reply:
Aha! That's an excellent find. We did make a pass and remove them all at some point, but they seem to have come back. I'm ashamed to admit that I reviewed some of those files and didn't notice it.